examples/presentation_definition_sd_jwt_vc.json

```diff
@@ -0,0 +1,42 @@
+{
+    "id": "d76c51b7-ea90-49bb-8368-6b3d194fc131",
+    "input_descriptors": [
+        {
+            "id": "IdentityCredential",
+            "format": {
+                "vc+sd-jwt": {
+                    "sd-jwt-alg": [
+                        "ES256"
+                    ],
+                    "kb-jwt-alg": [
+                        "ES256"
+                    ]
+                }
+            },
```

Comment:
We can put this information into the `vp_formats` metadata parameter. However, I'm no longer convinced that this is the best way to do it simply because it moves something that belongs to the request of a certain artifact to a different structure losing the context. This means the verifier cannot differentiate what algorithms it support for a certain kind of credential. I don't see an advantage in putting this data into the client metadata. 

I don't buy into the dependency argument. We use PE, so we should PE for what is intended to be used for. Would you state the same for OAuth?

Reply:
Credential request time is the wrong time to select algorithms because it will result in avoidable inexplicable user-visible errors.

Imagine a scenario where the verifier supports algorithms A and B and the Wallet supports algorithms B and C.  If algorithm knowledge is exchanged at discovery time (before the credential request), the parties will know to use algorithm B.  Whereas if a request to use algorithm A is sent (the verifier not having discovered the wallet's supported algorithms in advance), then the entire request will fail with an "unsupported signing algorithm" error that the end-user won't be able to do anything about.

Discovery of algorithms in advance and specifying an algorithm at request time are not equivalent in terms of facilitating good user experiences.

Reply:
I understand and agree with the idea to discover/negotiate algorithms ahead of request time and that works well for OpenID Connect and OpenID4VCI. However in the case of OID4VP, the verifier typically does not register with the wallet in advance, in case of a custom scheme invocation it doesn't even know the wallet (provider). So the verifier's metadata is passed in the `client_metadata` request parameter. That means we are taking about whether the format/alg values are conveyed in the `presentation_definition` or `client_metadata` request parameter. 

Having said that, I checked OId4VP again and the `vp_formats` metadata parameter was defined to convey this kind of data. So I will stick to that design and modify the PR accordingly. I would like to point out that the examples given in OID4VP will need to be adopted, too (https://openid.bitbucket.io/connect/openid-4-verifiable-presentations-1_0.html#name-vc-signed-as-a-jwt-not-usin).

Reply:
please open a separate issue in OID4VP if needed. this PR/HAIP itself is probably not a place to make this change.

Reply:
I agree. I changed the PR to use the current approach. 
Filed [issue](https://github.com/openid/OpenID4VP/issues/53)